tests/validation/tests/v3_api/test_node_label.py

```diff
@@ -923,43 +924,43 @@ def create_project_client(request):
     cluster_node_template["node_pools"] = node_pools[0]
     cluster_node_template["test_label"] = test_label
     cluster_node_template["label_value"] = label_value
-
-    def fin():
-        client = get_user_client()
-        cluster = cluster_node_template["cluster"]
-        if cluster is not None:
-            node_pools_list = client.list_nodePool(clusterId=cluster.id).data
-            # get unique node template ids
-            client.delete(cluster_node_template["cluster"])
-            wait_for_cluster_delete(client, cluster["name"])
+    if RANCHER_CLEANUP_CLUSTER:
```

Comment:
It would be cleaner to have the conditional wrap the `request.addfinalizer(fin)` statement.

Reply:
Good call, I updated the conditional.